pyo3-macros-backend/src/pymethod.rs

```diff
@@ -1429,15 +1486,21 @@ impl SlotFragmentDef {
         let fragment_trait = format_ident!("PyClass{}SlotFragment", fragment);
         let method = syn::Ident::new(fragment, Span::call_site());
         let wrapper_ident = format_ident!("__pymethod_{}__", fragment);
-        let arg_types: &Vec<_> = &arguments.iter().map(|arg| arg.ffi_type(ctx)).collect();
-        let arg_idents: &Vec<_> = &(0..arguments.len())
-            .map(|i| format_ident!("arg{}", i))
-            .collect();
+
         let mut holders = Holders::new();
-        let body = generate_method_body(
+        let MethodBody {
+            arg_idents,
+            arg_types,
+            trampoline: _,
+            body,
+        } = generate_method_body(
             cls,
             spec,
-            arguments,
+            &SlotCallingConvention::Fixed {
+                // the func_ty is not used here
+                func_ty: StaticIdent::new(""),
```

Comment:
It feels a bit weird to me that we rely on this just getting passed through here...
Maybe that suggests that this `func_ty` does not need to be part of this variant?

Reply:
Agreed, let me see if I can clean this up further 👍

Reply:
I pushed eba03f3e2aed4855fb946a113e0be6c005ec4435 which takes `func_ty` out of the enum and does some general cleanup to how these things are defined.